src/interfaces/vault.ts

```diff
@@ -263,22 +264,29 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
     return await Promise.all(
       adapters.map(async adapter => {
         const tokenAddresses = await adapter.tokens(overrides);
-        const tokens = await this.yearn.services.helper.tokens(tokenAddresses, overrides);
         const icons = this.yearn.services.asset.icon(tokenAddresses.concat(EthAddress));
-        const tokensMetadata = await this.yearn.tokens.metadata(tokenAddresses);
+        const tokensPromise = this.yearn.services.helper.tokens(tokenAddresses, overrides);
+        const tokensMetadataPromise = this.yearn.tokens.metadata(tokenAddresses);
+
+        const [tokens, tokensMetadata] = [await tokensPromise, await tokensMetadataPromise];
```

Comment:
Can use await Promise.all here too to fetch both async

Reply:
I thought this did the same thing as that? So they're both run in parallel

Reply:
I think there is no equivalent yet for Promise.all in async/await syntax.

Just for reference:
https://stackoverflow.com/questions/45285129/any-difference-between-await-promise-all-and-multiple-await

Reply:
So basically the benefit is just when failing, which resolves faster. But should not affect the happy path then.